src/types/Props.ts

```diff
@@ -0,0 +1,38 @@
+import { Todo } from './Todo';
+import { FilterStates } from '../types/enums';
+
+export interface ErrorNotificationProps {
+  error: string | null;
+  onHideError: () => void;
+}
+
+export interface FooterProps {
+  todos: Todo[];
+  filter: FilterStates;
+  setFilter: React.Dispatch<React.SetStateAction<FilterStates>>;
+  onClearCompleted: () => void;
+}
+
+export interface HeaderProps {
+  allCompleted: boolean;
+  onAddTodo: (title: string) => Promise<void>;
+  isLoading: boolean;
+  setError: (error: string | null) => void;
+  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>;
+  title: string;
+  setTitle: React.Dispatch<React.SetStateAction<string>>;
+}
+
+export interface TodoItemProps {
+  todo: Todo;
+  onDelete: (todoId: number) => Promise<void>;
+  isLoading: boolean;
+}
+
+export interface TodoListProps {
+  todos: Todo[];
+  onDelete: (todoId: number) => Promise<void>;
+  isLoading: boolean;
+  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>;
+  isDeleting: boolean;
+}
```

Comment:
I don't think it is necessary to put the `Props` in a separate file

Reply:
yep, u don't need it. 
it is better to keep props in the component you need, it is more convenient